app/models/child.rb

```diff
@@ -165,6 +165,154 @@ def initialize *args
                     }
                  }
               }"
+
+    view :by_followup_date,
+    :map => "function(doc) {
+					if (doc['couchrest-type'] == 'Child') {
+						if (doc.hasOwnProperty('gbv_follow_up_subform_section') && doc['gbv_follow_up_subform_section'] && doc['gbv_follow_up_subform_section'].length) {
+							doc['gbv_follow_up_subform_section'].forEach(function(followUp) {
+								if (followUp['followup_date']) {
+									emit([doc['owned_by_agency'], followUp['followup_date']], null);
+								}
+							});	
+						}
+					}
+				}"
+
+    view :by_days_between_case_opening_and_closure,
+    :map => "function(doc) {
+        if (doc['couchrest-type'] == 'Child' && doc.hasOwnProperty('date_closure') && doc['date_closure'] != null) {
+          var dateOpen = new Date(doc['registration_date']);
+          var dateClosure = new Date(doc['date_closure']);
+          var oneDay = 1000 * 60 * 60 * 24;
+          var dayDifference = +((dateClosure - dateOpen) / oneDay).toFixed(0)
+          var dayRange = #{Kpi::CASE_CLOSURE_TYPE[:days_ranges]}
+            .filter(function (ranges) {
+              return dayDifference >= ranges[0] && dayDifference <= ranges[1]
+            });
+          emit([doc['owned_by_agency'], dayRange[0] ? dayRange[0][0] + '-' + dayRange[0][1] : 'other'], null)
+          emit([doc['owned_by_agency'], '_count'], null)
+        }
+      }",
+    :reduce => "function(key, values, rereduce) {
+        if (rereduce) {
+          return sum(values);
+        }
+        return values.length
+      }"
+
+    view :by_days_between_case_opening_and_closure_high_risk,
+    :map => "function(doc) {
+        if (
+              doc['couchrest-type'] == 'Child' &&
+              doc.hasOwnProperty('date_closure') &&
+              doc['date_closure'] != null &&
+              doc.hasOwnProperty('risk_level') &&
+              doc['risk_level'] == 'high'
+            ) {
+          var dateOpen = new Date(doc['registration_date']);
+          var dateClosure = new Date(doc['date_closure']);
+          var oneDay = 1000 * 60 * 60 * 24;
+          var dayDifference = +((dateClosure - dateOpen) / oneDay).toFixed(0)
+          var dayRange = #{Kpi::CASE_CLOSURE_TYPE[:days_ranges]}
+            .filter(function (ranges) {
+              return dayDifference >= ranges[0] && dayDifference <= ranges[1]
+            });
+          emit([doc['owned_by_agency'], dayRange[0] ? dayRange[0][0] + '-' + dayRange[0][1] : 'other'], null)
+          emit([doc['owned_by_agency'], '_count'], null)
+        }
+      }",
+    :reduce => "function(key, values, rereduce) {
+        if (rereduce) {
+          return sum(values);
+        }
+        return values.length
+      }"
+
+    view :by_closure_reason,
```

Comment:
I think most if not all of these new views could be better implemented using the existing report logic which queries solr.  The existing report logic provides the capability to filter, group dates by a range, etc